### PR TITLE
Add task for fetching subscription details

### PR DESCRIPTION
**Pull request by flozia**
## Description

This PR adds a new task for fetching subscription and payment information.

## Reference

- [VPN-2067](https://mozilla-hub.atlassian.net/browse/VPN-2067)
- [PSP-370](https://mozilla-hub.atlassian.net/jira/software/c/projects/VPN/boards/344?modal=detail&selectedIssue=PSP-370)

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  m_authenticationInAppSession = new AuthenticationInAppSession(this);
+
+  connect(m_authenticationInAppSession, &AuthenticationInAppSession::terminated,
+          this, &Task::completed);
+
+  connect(m_authenticationInAppSession, &AuthenticationInAppSession::completed,
+          this, [this, pkceCodeVerifier](const QString& pkceCodeSuccess) {
+            logger.debug() << "Authentication completed with code:"
+                           << pkceCodeSuccess;
```
Comment (bakulf) [inline src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp:87]: logger.`debug.sensitive()` here and in the authenticate.cpp - we don't need to write it in prod (it's an existing bug, but let's fixing in this PR)

File: src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            NetworkRequest* request =
+                NetworkRequest::createForAuthenticationVerification(
+                    this, pkceCodeSuccess, pkceCodeVerifier);
+
+            connect(
+                request, &NetworkRequest::requestFailed, this,
+                [this](QNetworkReply::NetworkError error, const QByteArray&) {
+                  logger.error()
+                      << "Failed to complete the authentication" << error;
+                  m_authenticationInAppSession->terminate();
```
Comment (bakulf) [inline src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp:98]: What does it happen to this `Task` when this happens? We need to `complete` it somehow.

Comment (bakulf) [reply src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp:98]: What about if you connect a method to `NetworkRequest::destroyed` and you check the logic there?

**Review by bakulf — COMMENTED** [review]
I have read only the network request part and it looks really really good! Well done!

File: src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp (hunk 2 above)
Comment (flozia) [reply src/tasks/getsubscriptiondetails/taskgetsubscriptiondetails.cpp:98]: If the auth session it terminated completing the `Task` should be handled by `connect(m_authenticationInAppSession, &AuthenticationInAppSession::terminated, this, &Task::completed);` ([L81-82](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/3538/files#diff-0de758d86f6de3f1fc61b369c831dfc6c1f3e4458dff91ab3a4312f604b2f8c6R81-R82)) if I’m not mistaken?

File: src/mozillavpn.cpp [hunk 3]
```diff
@@ -712,6 +713,14 @@ void MozillaVPN::serversFetched(const QByteArray& serverData) {
   }
 }
 
+void MozillaVPN::subscriptionDetailsFetched(
+    const QByteArray& subscriptionDetailsData) {
+  logger.debug() << "Subscription details data fetched!";
+  Q_UNUSED(subscriptionDetailsData);
+
+  requestSubscriptionManagement();
```
Comment (bakulf) [inline src/mozillavpn.cpp:721]: Does the loading view have a cancel button/link? if yes, we don't know if we are in the right state here

File: src/mozillavpn.cpp [hunk 4]
```diff
@@ -712,6 +713,14 @@ void MozillaVPN::serversFetched(const QByteArray& serverData) {
   }
 }
 
+void MozillaVPN::subscriptionDetailsFetched(
+    const QByteArray& subscriptionDetailsData) {
+  logger.debug() << "Subscription details data fetched!";
+  Q_UNUSED(subscriptionDetailsData);
```
Comment (bakulf) [inline src/mozillavpn.cpp:719]: Add a TODO

File: src/mozillavpn.cpp (hunk 3 above)
Comment (bakulf) [reply src/mozillavpn.cpp:721]: Plus, why not calling directly `emit subscriptionManagementNeeded()`? You can get rid of `requestSubscriptionManagement` it seem.

File: src/mozillavpn.cpp [hunk 5]
```diff
@@ -1387,6 +1396,13 @@ void MozillaVPN::requestContactUs() {
   emit contactUsNeeded();
 }
 
+void MozillaVPN::requestSubscriptionManagement() {
+  logger.debug() << "Subscription management view requested";
+
+  QmlEngineHolder::instance()->showWindow();
```
Comment (bakulf) [inline src/mozillavpn.cpp:1402]: Why this line?

File: src/mozillavpn.cpp [hunk 6]
```diff
@@ -1767,3 +1783,9 @@ void MozillaVPN::cancelAccountDeletion() {
   logger.warning() << "Canceling account deletion";
   AuthenticationInApp::instance()->terminateSession();
 }
+
+void MozillaVPN::getSubscriptionDetails() {
+  logger.debug() << "Get subscription details";
+  TaskScheduler::scheduleTask(
```
Comment (bakulf) [inline src/mozillavpn.cpp:1789]: Do you want to check the `m_state` first?

**Review by bakulf — CHANGES_REQUESTED** [review]
If you are not planning to land the QML part, remove the changes in MozillaVPN because it's hard to guess how those methods/signals will be used

**Comment by flozia** [comment]
I reverted the changes in  MozillaVPN as suggested and decided to land the QML and rest that is needed for the Subscription management view in a separate PR.

**Pull request merged**